### PR TITLE
Azure.Monitor.Query: invalid date string format in MetricsClient

### LINKED ISSUE #40047: [BUG] QueryResourceAsync causing error if used with TimeRange of QueryTimeRange(Datetime, Datetime)
### Library name and version

Azure.Monitor.Query 1.2.0

### Describe the bug

Using **MetricsQueryClient.QueryResourceAsync()** with option **TimeRange** of **QueryTimeRange(startTime, endTime)** causes and error, opposed to TimeRange of QueryTimeRange(duration).

***Exception or Stack Trace***
'Detected invalid time interval input: 2023-11-15T10:12:25.4174057 01:00/2023-11-15T11:12:25.4174057 01:00, supported Iso 8601 time interval format: (Datetime/Datetime, Datetime/Duration, Duration/Datetime, Duration)
Status: 400 (Bad Request)
ErrorCode: BadRequest

### Expected behavior

Receive the same result as if used with

```
var options = new MetricsQueryOptions
{
    Aggregations = { MetricAggregationType.Average },
    Granularity = TimeSpan.FromMinutes(1),
    TimeRange = new QueryTimeRange(TimeSpan.FromHours(1))
};
```

### Actual behavior

Receiving Status: 400 (Bad Request)

### Reproduction Steps

```
var metricsQueryClient = new MetricsQueryClient(new DefaultAzureCredential());
var endTime = DateTimeOffset.Now;
var startTime = endTime.AddHours(-1);

var resourceId = <someAppServiceResourceId>;
var metricsNames  = new List<string> { "HttpResponseTime" };
var options = new MetricsQueryOptions
{
    Aggregations = { MetricAggregationType.Average },
    Granularity = TimeSpan.FromMinutes(1),
    TimeRange = new QueryTimeRange(startTime, endTime)
};

var result = await metricsQueryClient.QueryResourceAsync(resourceId, metricsNames, options);
```

### Environment

- Windows 10

- VisualStudio 17.6.4

- .NET 7.0

- Azure.Monitor.Query 1.2.0

### PR DESCRIPTION
**Pull request by wps0**
Fixes #40047

### CODE REVIEW

**Comment by wps0** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by nisha-bhatia** [comment]
Hi @wps0, thank you for your contribution. Could you please add a test showing this change? Thank you.

File: sdk/monitor/Azure.Monitor.Query/src/MetricsClientExtensions.cs [hunk 1]
```diff
@@ -51,5 +52,20 @@ internal static IList<string> CommaSplit(string value) =>
                 new List<string>() :
                 // TODO: #10600 - Verify we don't need to worry about escaping
                 new List<string>(value.Split(','));
+
+        internal static string ToIsoString(this DateTimeOffset value)
```
Comment (nisha-bhatia) [inline sdk/monitor/Azure.Monitor.Query/src/MetricsClientExtensions.cs:56]: @wps0 Why are we adding this method to the extensions file as well?

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientLiveTests.cs [hunk 2]
```diff
@@ -707,7 +707,8 @@ public async Task CanSetServiceTimeout()
                     // or a partial failure 200 response
                     if (exception.Status == 200)
                     {
-                        StringAssert.Contains("Query cancelled by the user's request", exception.Message);
+                        //StringAssert.Contains("Query cancelled by the user's request", exception.Message);
```
Comment (nisha-bhatia) [inline sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientLiveTests.cs:710]: TODO: confirm before merging that error message comparison is okay

Comment (nisha-bhatia) [reply sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientLiveTests.cs]: Partial failure message has just changed so I think this is fine as it still checks for the same error.

File: sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs [hunk 3]
```diff
@@ -150,8 +150,13 @@ private async Task<Response<MetricsQueryResourcesResult>> ExecuteBatchAsync(IEnu
...
                 if (options.TimeRange != null)
                 {
-                    startTime = options.TimeRange.Value.Start.ToString();
-                    endTime = options.TimeRange.Value.End.ToString();
+                    if (options.TimeRange.Value.Start == null || options.TimeRange.Value.End == null)
+                    {
+                        throw new ArgumentNullException("Both start and end times must be set in the TimeRange.");
+                    }
+
+                    startTime = options.TimeRange.Value.Start.ToIsoString();
```
Comment (nisha-bhatia) [inline sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs:158]: Updated this logic to not use duration as the service does not accept that value. StartTime and EndTime must both be set as the service takes both.

**Comment by nisha-bhatia** [comment]
Follow-up work item: https://github.com/Azure/azure-sdk-for-net/issues/46454

**Review by nisha-bhatia — APPROVED** [review]

**Review by Thompson1985 — APPROVED** [review]

**Pull request merged**